### PR TITLE
showgraph: rework output directory use and defaults

### PR DESCRIPTION
**Pull request by kmantel**
- change subdirectory from "show_graph OUTPUT" or "show_graph output" to "pnl-show_graph-output"
- use pathlib to make directory/file names
- avoid using python site-packages dir as a default if psyneulink is installed as non-editable

### CODE REVIEW

**Comment by kmantel** [comment]
related to issue in #3085

File: psyneulink/core/compositions/showgraph.py [hunk 1]
```diff
@@ -288,6 +291,32 @@
...
+        site_packages_dirs = []
+
+    # if psyneulink is installed in site-packages (not local/editable),
+    # don't put show_graph files there
+    for d in site_packages_dirs:
+        if pathlib.Path(d) in pnl_module_dir.parents:
+            default_dir = pathlib.Path('.')
+            break
+    else:
+        default_dir = pnl_module_dir.joinpath('..')
```
Comment (jvesely) [inline psyneulink/core/compositions/showgraph.py:313]: Can this use `pnl_module_dir.parent` ?

**Review by jvesely — APPROVED** [review]

File: psyneulink/core/compositions/showgraph.py [hunk 2]
```diff
@@ -288,6 +291,32 @@
...
+
+default_showgraph_subdir = 'pnl-show_graph-output'
+
+
+def get_default_showgraph_dir():
+    pnl_module_dir = pathlib.Path(psyneulink.__file__).parent.absolute()
+    try:
+        site_packages_dirs = site.getsitepackages()
+    except AttributeError:
+        # virtualenv <20 overrides site and has no getsitepackages
```
[the remark this answers is not in the document]
Comment (kmantel) [reply psyneulink/core/compositions/showgraph.py:303]: I don't know about other envs - I just happened across that specific issue online, not personally. I use pyenv, and it's ok for that.

I'm not familiar with direct-url but looked into it a bit, and looking at my own python env, I'm not sure I can get reliable info out of it. Some of my editable installs don't have a direct_url.json, and others do. My psyneulink editable install for example doesn't appear to have a .dist-info folder at all. Only a couple other non-editable installs out of >200 packages also have direct_url.json

I'm open to other options, but I suspect checking site-packages should handle common cases.

File: psyneulink/core/compositions/showgraph.py (hunk 1 above)
Comment (kmantel) [reply psyneulink/core/compositions/showgraph.py:313]: Yes, changed

File: psyneulink/core/compositions/showgraph.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jvesely) [reply psyneulink/core/compositions/showgraph.py:303]: makes sense. thanks for checking.

**Pull request merged**